lib/Dialect/Secret/IR/SecretPatterns.cpp

```diff
@@ -16,7 +18,19 @@ LogicalResult CollapseSecretlessGeneric::matchAndRewrite(
     }
   }
 
-  YieldOp yieldOp = dyn_cast<YieldOp>(op.getBody()->getOperations().back());
+  // Allocation ops cannot be collapsed because they may be used in later
+  // generics. We should eventually do a proper dataflow analysis to ensure
+  // they can be collapsed when no secret data is added to them.
+  //
+  // There is no good way to identify an allocation op in general. Maybe we can
+  // upstream a trait for this?
+  for (auto &op : op.getBody()->getOperations()) {
```

Comment:
I think you could also use `getOps<memref::AllocOp>()` and that way skip checking other ops

Reply:
Fixed! TIL about `[[maybe_unused]]`